common/flatpak-dir.c

```diff
@@ -1970,18 +1970,29 @@ _flatpak_dir_ensure_repo (FlatpakDir   *self,
           GKeyFile *orig_config = NULL;
           g_autoptr(GKeyFile) new_config = NULL;
           g_autofree char *orig_min_free_space_percent = NULL;
+          g_autofree char *orig_min_free_space_size = NULL;
           g_autofree char *orig_add_remotes_config_dir = NULL;
 
           orig_config = ostree_repo_get_config (repo);
           orig_min_free_space_percent = g_key_file_get_value (orig_config, "core", "min-free-space-percent", NULL);
+          orig_min_free_space_size = g_key_file_get_value (orig_config, "core", "min-free-space-size", NULL);
           orig_add_remotes_config_dir = g_key_file_get_value (orig_config, "core", "add-remotes-config-dir", NULL);
 
-          if (orig_min_free_space_percent == NULL ||
+          if (orig_min_free_space_size == NULL ||
               orig_add_remotes_config_dir == NULL)
```

Comment:
This condition needs to include `|| orig_min_free_space_percent != NULL`, otherwise `new_config` will still be set to `NULL` if the `if (orig_min_free_space_percent != NULL && g_ascii_strtoull (…) == 0)` branch is taken below.